### DISCUSSION TITLE
Is it possible to implement own connectors for custom protocols (like UMB)?

### DISCUSSION THREAD

**Discussion by mikedolx** [thingsboard/thingsboard#11369] [Q&A]
Hi,

I'm currently evaluating Thingsboard.io as an alternative to our IoT solution. We have a lot of sensors, that support the serial UMB  protocol. Unfortunately there is only a modbus connector available for the IoT gateway. Is it possible to write your own connector for this protocol? Are there maybe third-party-connectors available?

Thanks and BR,
Michael;

**Comment by K0I05** [comment]
UMB is an open protocol by Lufft (formerly, they were bought), it is somewhat comparable to MODBUS, and both protocols are classified as legacy protocols.  You could wrap the UMB message within an MQTT JSON message and then use a data converter (integration center).  However, I would leverage your gateway and translate the UMB message to a friendlier MQTT message and treat each sensor as a device.
